operations/helm/charts/mimir-distributed/templates/metamonitoring/grafana-agent.yaml

```diff
@@ -42,7 +46,18 @@ spec:
       securityContext:
         {{- . | toYaml | nindent 8 }}
   {{- end }}
-
+  {{- with .nodeSelector }}
+  nodeSelector:
+    {{- toYaml . | nindent 4 }}
+  {{- end }}
+  {{- with .tolerations }}
+  tolerations:
+    {{- toYaml . | nindent 4 }}
+  {{- end }}
+  {{- with .topologySpreadConstraints }}
+  topologySpreadConstraints:
+    {{- toYaml . | nindent 4 }}
+  {{- end }}
```

Comment:
Is there a reason why you haven't used the `topologySpreadConstraints` helper defined [here](https://github.com/grafana/mimir/blob/main/operations/helm/charts/mimir-distributed/templates/lib/topologySpreadConstraints.tpl)? [This](https://github.com/grafana/mimir/blob/c667ebefc5b347f04765d3180c9578941ac7f2e9/operations/helm/charts/mimir-distributed/templates/querier/querier-dep.yaml#L148) is an example of how it is used.

This will automatically inject the correct `labelSelector`, amongst other things.

Reply:
Ah thanks for the pointer @charleskorn. Applied that now